AI/Utilities/RemovalLogic.py

```
@@ -399,10 +399,52 @@ def battlefield_zone_ids(full_state: dict) -> set[int]:
+            if not isinstance(zone, dict):
+                continue
+            is_bf = zone.get("type") == "ZoneType_Battlefield" or (
+                known_zone_ids is not None and zone.get("zoneId") in known_zone_ids
+            )
+            if not is_bf:
+                continue
+            seen_zone = True
+            for inst_id in zone.get("objectInstanceIds", []) or []:
+                if isinstance(inst_id, int):
+                    ids.add(inst_id)
+    except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)